ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/client/Web3JExecutionEngineClient.java

```diff
@@ -122,6 +126,24 @@ private static PowBlock eth1BlockToPowBlock(EthBlock.Block eth1Block) {
     return doRequest(web3jRequest);
   }
 
+  private void handleError(String error) {
+    if (isErrored.compareAndSet(false, true)) {
+      EVENT_LOG.executionClientIsOffline(error);
+    }
+  }
+
+  private void handleSuccess() {
+    if (isErrored.compareAndSet(true, false)) {
+      EVENT_LOG.executionClientIsOnline();
+    }
+  }
+
+  private <T> SafeFuture<T> doWeb3JRequest(CompletableFuture<T> web3Request) {
+    return SafeFuture.of(web3Request)
+        .catchAndRethrow(error -> handleError(error.toString()))
```

Comment:
I'd just pass this as a `Throwable` rather than converting to String.  The `EventLog` can choose what to do with it but logging it as an exception is probably a good idea - it only logs when first switching to error state and the stack trace is omitted from the console output by default (it only goes to the log file).

Reply:
> stack trace is omitted from the console output by default (it only goes to the log file).

oh! ok then!